Series of small WCOSS2 updates - round 3

KateFriedman-NOAA opened the pull request:
This PR includes the following WCOSS2 updates:

1. add NODESIZE to setup scripts and hardcode to use on only WCOSS2 for now (value=128)
2. WCOSS2.env updates for the gldas, efcs, and waveawipsbulls jobs
3. add `memory_wafsgcip="50GB"` to config.resources
4. setup_expt bug fixes for waveinit/waveprep jobs (missing syntax from python2 to python3 conversion)
5. convert gfs_bufr, fbwndgfs, fv3nc2nemsio, ,regrid_nemsio, storm_reloc, enkf_chgres_recenter, and gaussian_sfcanl builds to LUA module format
6. fix for adding awips jobs to xml when DO_AWIPS=YES (fix by @WalterKolczynski-NOAA )

Refs: #399

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:feature/ops-wcoss2' into feature/ops-wcoss2

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:feature/ops-wcoss2' into feature/ops-wcoss2

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:feature/ops-wcoss2' into feature/ops-wcoss2

KateFriedman-NOAA committed:
GLDAS WCOSS2 env file updates

- add USE_CFP=YES for GLDAS CFP usage on WCOSS2
- update APRUN_GAUSSIAN launcher command for WCOSS2 format

Refs: #399

KateFriedman-NOAA committed:
Convert gfs_bufr build to LUA

- convert gfs_bufr WCOSS2 modulefile to LUA format
- update build_gfs_bufrsnd.sh for loading module

Refs: #399

KateFriedman-NOAA commented:
I will be adding a few more changes to this PR today (at least the remaining LUA format changes).

aerorahul commented on ush/rocoto/rocoto.py:
```diff
@@ -68,6 +68,7 @@ def create_task(task_dict):
     jobname = task_dict.get('jobname', 'demojob')
     account = task_dict.get('account', 'batch')
     queue = task_dict.get('queue', 'debug')
+    nodesize = task_dict.get('nodesize', '128')
```
Can we set this default to 1?

WalterKolczynski-NOAA replied:
Not sure it matters. This branch is only going to be used for ops. And at any rate, usually defaults are to the ops configuration.

KateFriedman-NOAA replied:
Changed default to 1, thanks!

aerorahul reviewed (commented):
All changes are good.
Just 1 suggestion.  See below.

KateFriedman-NOAA committed:
Update fbwndgfs build for LUA format

- update fbwndgfs modulefile to LUA format
- update build_gfs_fbwndgfs.sh for module load command

Refs: #399

WalterKolczynski-NOAA commented on ush/rocoto/setup_workflow.py:
```diff
@@ -197,6 +197,8 @@ def get_definitions(base):
         strings.append(f'''\t<!ENTITY PARTITION_SERVICE "{base['QUEUE_SERVICE']}">\n''')
     strings.append(f'\t<!ENTITY SCHEDULER  "{scheduler}">\n')
+    if machine in ['WCOSS2']:
+       strings.append(f'\t<!ENTITY NODESIZE  "128">\n')
```
Should this be based on the nodesize set above?

KateFriedman-NOAA replied:
It should...this is just hardcoded for ease/time. See comment below in PR.

WalterKolczynski-NOAA commented on ush/rocoto/setup_workflow_fcstonly.py:
```diff
@@ -130,6 +130,8 @@ def get_definitions(base):
        strings.append(f'''\t<!ENTITY PARTITION_SERVICE "{base['QUEUE_SERVICE']}">\n''')
     strings.append(f'\t<!ENTITY SCHEDULER  "{scheduler}">\n')
+    if machine in ['WCOSS2']:
+       strings.append(f'\t<!ENTITY NODESIZE  "128">\n')
```
Should this be based on the nodesize set above?

KateFriedman-NOAA replied:
It should...this is just hardcoded for ease/time. See comment below in PR.

aerorahul replied:
What is the else value for this `NODESIZE`.
If some one works with this branch on orion or any other, this will fail to put a `NODESIZE` value.  It is required now in the task.

KateFriedman-NOAA replied:
Can one of you guys help finish adding NODESIZE to the setup scripts then? I don't think I'm going to have time before hand-off. If it's not remedied before hand-off then it can be sorted out in the feature/ops-orion branch when I update that (that's where we currently support ops on the R&Ds).

KateFriedman-NOAA replied:
Incorporated @aerorahul patch, thanks! That gets us most of the way there and removes the WCOSS2 hardcoding. We will just need to get it reading from the env files later.

KateFriedman-NOAA committed:
Update fv3nc2nemsio build for LUA format

- update WCOSS2 fv3nc2nemsio module for LUA format
- update build_fv3nc2nemsio.sh to load module instead of sourcing

Refs: #399

KateFriedman-NOAA committed:
Update regrid_nemsio build to LUA format

- update WCOSS2 regrid_nemsio modulefile to LUA format
- update build_regrid_nemsio.sh to load module instead of sourcing

Refs: #399

KateFriedman-NOAA committed:
Update storm_reloc build to LUA format

- update storm_reloc WCOSS2 modulefile to LUA format
- update build_tropcy_NEMS.sh to load module instead of source

Refs: #399

KateFriedman-NOAA committed:
Update enkf_chgres_recenter and gaussian_sfcanl build to LUA format

- update enkf_chgres_recenter and gaussian_sfcanl modulefiles to LUA
format
- update build_enkf_chgres_recenter.sh and build_gaussian_sfcanl.sh to
load module instead of sourcing

Refs: #399

KateFriedman-NOAA commented:
@WalterKolczynski-NOAA @aerorahul I have committed the remaining WCOSS2 LUA format modulefile updates that needed doing. Please also see those changes in this PR now.

Regarding NODESIZE in the setup scripts...the addition is not complete. For time and quick testing I did the latter half of the work to add NODESIZE into the rocoto xmls on WCOSS2. I hardcoded it for just WCOSS2 for now but need to do the first half of the work still = have NODESIZE/nodesize set based on the machine (parse the value from the env file or configs?). Will need to finish this on the dev side (if no time to finish for ops).

WalterKolczynski-NOAA commented:
> @WalterKolczynski-NOAA @aerorahul I have committed the remaining WCOSS2 LUA format modulefile updates that needed doing. Please also see those changes in this PR now.
> 
> Regarding NODESIZE in the setup scripts...the addition is not complete. For time and quick testing I did the latter half of the work to add NODESIZE into the rocoto xmls on WCOSS2. I hardcoded it for just WCOSS2 for now but need to do the first half of the work still = have NODESIZE/nodesize set based on the machine (parse the value from the env file or configs?). Will need to finish this on the dev side (if no time to finish for ops).

My issue is just that it is hardcoded in several places, instead of just one.

KateFriedman-NOAA committed:
Change nodesize default to 1 in rocoto.py

Refs: #399

KateFriedman-NOAA committed:
Moved COMINsyn setting to vrfy.sh from config

- move COMINsyn setting from config.vrfy to vrfy.sh
- avoids message from setup_workflow*.py when it runs and invokes the
compath.py script being used in the default COMINsyn setting
- now gets defined in vrfy.sh before use with VRFYTRAK task

Refs: #399

KateFriedman-NOAA commented:
> My issue is just that it is hardcoded in several places, instead of just one.

It's now hardcoded only once for each setup_workflow*.py script:
```
    if machine in ['WCOSS2']:
       strings.append(f'\t<!ENTITY NODESIZE  "128">\n')
```
If I find time I will try to get it parsing from a file and un-hardcode it. Will definitely resolve in the dev work if not for ops (where it isn't used anyway).

KateFriedman-NOAA commented:
@aerorahul @WalterKolczynski-NOAA I've committed the final changes I wanted to get into this PR (e.g. final LUA format conversions). Please continue your reviews, thanks!

aerorahul commented:
> > My issue is just that it is hardcoded in several places, instead of just one.
> 
> It's now hardcoded only once for each setup_workflow*.py script:
> 
> ```
>     if machine in ['WCOSS2']:
>        strings.append(f'\t<!ENTITY NODESIZE  "128">\n')
> ```
> 
> If I find time I will try to get it parsing from a file and un-hardcode it. Will definitely resolve in the dev work if not for ops (where it isn't used anyway).

@KateFriedman-NOAA 
Please apply this patch.  I have not checked the nodesizes for all machines, since I don't know that off the top of my head.
This should atleast keep this generic.
[diff.txt](https://github.com/NOAA-EMC/global-workflow/files/7803753/diff.txt)

WalterKolczynski-NOAA commented:
npe_node_max is defined in the env file for every machine. Why are we not using that?

aerorahul commented:
> npe_node_max is defined in the env file for every machine. Why are we not using that?

We should grab the value for `nodesize` from `npe_node_max`, but it will require more work to make it available in the function `get_definitions`.  I leave it to you to accept/reject the patch.

KateFriedman-NOAA committed:
NODESIZE and NATIVE update to setup_workflow

- update NODESIZE setting to support all currently supported machines
- add vscatter NATIVE setting for WCOSS2
- fix small syntax bug

Refs: #399

KateFriedman-NOAA commented:
Committed the NODESIZE patch (tested, works) and included a NATIVE update to add `-l place=vscatter` into the rocoto NATIVE xml settings. Also fixed a small syntax error (an errant f caused it to fail).

KateFriedman-NOAA committed:
Update to natstr in workflow_utils.py

- remove "f" in new natstr line for WCOSS2, not needed

Refs: #399

WalterKolczynski-NOAA committed:
Update nodesize to be read from config.resources

Instead of defining the nodesize in a new place, the nodesize is
now read from the config.resources npe_node_max setting. To pass the
value to the XML generation, the value is taken from fcst and added
to base (the value should be the same for all tasks).

Refs: #399

KateFriedman-NOAA committed:
Fix error in awips group fhr line

- setup_workflow.py failed when DO_AWIPS=YES
- updated fhrs line to wrap it in line()
- fix made by Walter Kolczynski

Refs: #399

KateFriedman-NOAA commented:
Included fix for adding awips jobs to this PR, thanks for the fix @WalterKolczynski-NOAA ! Will now submit.
